### DISCUSSION TITLE
Calculating indicators for winter months

### DISCUSSION THREAD

**Discussion by TarekAMZ** [Ouranosinc/xclim#1793] [Questions] [answered]
### Setup Information

* Xclim version:

### Context

Is there a quick way to calculate e.g., the annual number of dry spell days for winter only, season = ['DFJ']? However, that is for the correct winter season. For example, for the year 1960, that would be December 1959 until end of February 1960. 

### Steps To Reproduce

```
$ pip install foo --bar
```

**Comment by coxipi** [comment]
Hi! What you're looking for is:
* `freq = QS-DEC` for correct winter season
* some `xclim` functions have a `**indexer` argument. You can give arguments for `xclim.core.calendar.select_time`, in your case you want to specify `month = [1,2,12]`. 

Here is an example:
```
import xclim 
from xclim.testing import open_dataset 
pr = open_dataset("ERA5/daily_surface_cancities_1990-1993.nc").isel(location=0).pr
xclim.atmos.dry_spell_total_length(pr, freq="QS-DEC",  month=[1,2,12])
```
Incomplete winters on the edges (e.g. data starting on 1990-01-01 -> incomplete '89-'90 winter)  and other seasons than winters are filled with NaNs. You can change the `window` for the number of days in a spell and `thresh` for the threshold defining dry days.

Cheers!

**Reply by TarekAMZ** [reply]
Hi, 
Thanks for the quick reply. In fact, I'm working with the dry spell total length indicator. However, I would get to the number of dry days after. 
I tried this solution and it worked fine. I also tried to drop the other months using 'drop = True' argument, but it did not make an effect, the other months remain as nan. I did a workaround by postprocessing the output by calculating the annual max, which is fine but not very efficient. If you have other ideas, that would be nice to know :) 
Thanks again,
Tarek

**Reply by coxipi** [reply, marked as answer]
Hmm, it seems that  `resample` in the indicator reintroduces times that were cut off, e.g. here is a minimal code showing this:
```python
sub_pr  = xclim.core.calendar.select_time(pr, month=[1,2,12], drop=True)
sub_pr.resample(time="QS-DEC").mean().time
```
times outside the winters are also present.

Since all times are masked outside the winter season, you can also use `YS-DEC`, now I think about it, it's really what you asked for in the beginning haha, should have thought of that directly! :
```python
xclim.atmos.dry_spell_total_length(pr, freq="YS-DEC", month=[1,2,12])
>>> [nan, 31., 22., 30., nan]
```
that gives the appropriate output.

Cheers!

**Reply by TarekAMZ** [reply]
Yes, this is exactly what I needed. Thanks a lot for the help :) 

Cheers!
